events/start_handler.go

```diff
@@ -92,7 +93,11 @@ func (h *StartHandler) Handle(event *docker.APIEvents) error {
 	if err != nil {
 		return err
 	}
+
 	if rancherIP == "" {
+		if _, ok := c.Config.Labels[RancherDns]; ok {
```

Comment:
Your not checking that it equals "true".  You can just do the following and go will be happy

```go
if c.Config.Labels[RancherDns] == "true" {
...
}
```